### PR TITLE
i18n(zh-cn): Update `server-side-rendering.mdx`

### PR DESCRIPTION
**Pull request by Nin3lee**
#### Description (required)

Update `server-side-rendering.mdx`
- add the missing docs
- fix the wrong headings

#### Related issues & labels (optional)

- #9700 
- Suggested label: `i18n` `hacktoberfest-accepted`

please add the label `hacktoberfest-accepted` after review, thanks for the reviewer!

### CODE REVIEW

File: src/content/docs/zh-cn/guides/server-side-rendering.mdx [hunk 1]
```diff
@@ -224,26 +224,71 @@ Astro.cookies.set('counter', String(counter))
 
 ### `Response`
 
-你也可以从使用按需渲染的任何页面返回一个 [响应](https://developer.mozilla.org/zh-CN/docs/Web/API/Response)。
+[`Astro.response`](/zh-cn/reference/api-reference/#astroresponse) 是一个标准 [`ResponseInit`](https://developer.mozilla.org/zh-CN/docs/Web/API/Response/Response#options) 对象。它可用于设置响应状态和标头。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/server-side-rendering.mdx:227]:
```suggestion
[`Astro.response`](/zh-cn/reference/api-reference/#astroresponse) 是一个标准 [`ResponseInit`](https://developer.mozilla.org/en-US/docs/Web/API/Response/Response#options) 对象。它可用于设置响应状态和标头。
```

外链如果对应的中文版本落后的话，就用英文原版

File: src/content/docs/zh-cn/guides/server-side-rendering.mdx [hunk 2]
```diff
@@ -254,7 +299,7 @@ if(!product) {
 在 `server` 和 `hybrid` 模式下，对于非静态生成的页面，你可以从这个对象中获取更多信息。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/server-side-rendering.mdx:299]: 上一行的 `请求` 改一下，不用翻译 `Request`

**Review by liruifengv — APPROVED** [review]
LGTM!

**Review by dreyfus92 — APPROVED** [review]
Thank you peeps!

**Pull request merged**